### PR TITLE
Implement traits for tuples even when there are no auto-traits

### PR DESCRIPTION
**Pull request by Ekleog**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -71,6 +71,12 @@ macro_rules! implement_marker_for {
     };
 }
 
+macro_rules! implement_marker_for_tuples {
+    (unsafe impl $trait:ident for $(($($i:ident),*)),*) => {
+        $(unsafe impl<$($i: $trait),*> $trait for ($($i),*) {})*
+    }
+}
```
Comment (nagisa) [inline src/lib.rs:82]: How about something like this instead:

```rust
macro_rules! implement_marker_for_tuples {
    (unsafe impl $trait:ident for) => { };
    (unsafe impl $trait:ident for $first_generic:ident, $($generic:ident,)* ) => {
        unsafe impl<$first_generic, $($generic),*> $trait for ($first_generic, $($generic),*)
        where $first_generic: $trait, $($generic: $trait),* {
        }
        implement_marker_for_tuples! { unsafe impl $trait for $($generic,)* }
    }
}

implement_marker_for_tuples! {
    unsafe impl ProcSend for T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15,
}
```

**Review by nagisa — COMMENTED** [review]
It would probably be good to increase the implementation counts up to 16-tuples or even 32-tuples. Exceeding 8-tuples seems relatively likely to happen.

**Comment by Ekleog** [comment]
Thank you for the recursive macro-based code cleanup idea!

About the number of 8, 16 or 32, I seemed to remember the standard library limited its default trait implementations to 8 but it's actually 12, so I've increased the number of tuple impls to 12 to match stdlib :)

**Comment by Ekleog** [comment]
r=@lovesegfault, I'll merge and bump the crates.io version to 0.1.1 when I get a chance

**Pull request merged**